### PR TITLE
Revise Chapter 6 and 7 for RC02

### PR DESCRIPTION
**Pull request by g1zzm0**
This is a revision of the data structures we defined previously for 
Chapter 6 and 7, but eventually removed as we, erroneously, 
deemed them as outside the core meta-model.

We:
* Update  Class DataSpecification to RC02
* Update  Class DataSpecificationContent to RC02
* Update  Class DataTypeIEC61360 to RC02
* Update Class DataSpecificationIEC61360 to RC02
* Add Class Category_Concept_Description
* Update Class DataSpecificationPhysicalUnit  to RC02
* Add Chapter Security

### CODE REVIEW

**Comment by g1zzm0** [comment]
review-requested: @sebbader

**Comment by g1zzm0** [comment]
review-requested: @Manu3756

**Comment by g1zzm0** [comment]
review-requested: @constantinziesche

**Comment by mristin** [comment]
@g1zzm0 nitpick: we have to write classes in `Pascal_case` (exception: abbreviations all uppercase except `id`). 

Example: `Data_specification_IEC_631160`

**Comment by mristin** [comment]
@g1zzm0 the enum `Category_Concept_Description` is indeed redundant. Since `category` is a string attribute, I introduced the constant set `Valid_concept_description_categories`.

**Comment by sebbader** [comment]
@g1zzm0 honestly, I have a hard time reviewing by the code, is there e.g. a JSON Schema output I can check?

Independently of that, two things from the last working group discussion:
* LangStingSets (https://github.com/aas-core-works/aas-core-meta/blob/4734fff56eef9251e95fc2ca03dbbc23fd0346cc/aas_core_meta/v3rc2.py#L4333) I assume there will be another PR for #128 ?
* HasDataSpecification/embeddedDataSpecification in https://github.com/aas-core-works/aas-core-meta/blob/4734fff56eef9251e95fc2ca03dbbc23fd0346cc/aas_core_meta/v3rc2.py#L1675 will be introduced also in another PR?

**Comment by g1zzm0** [comment]
@sebbader As discussed before, the schema generation is done after the validation of the metamodel or now the classes from chapter 6 and 7. The workflow will be Python MM Modeling -> Review MM -> Validation -> Code Generation -> New Schema Generation -> Check MM based on the serialization.  This is necessary to avoid unnecessary correction loops and thus delays.

**Comment by g1zzm0** [comment]
@sebbader for the potential change of HasDataSpecification see the branch https://github.com/aas-core-works/aas-core-meta/tree/g1zzm0/EmbeddedDataSpecification

File: aas_core_meta/v3rc2.py [hunk 1]
```diff
@@ -4530,36 +4689,48 @@ def __init__(
...
-    IEC 61360.
-    Although the IEC61360 attributes listed in this template are defined for properties
-    and values and value lists only it is also possible to use the template for other
-    definition This is shown in the tables Table 7, Table 8, Table 9 and Table 10.
+    Content of data specification template for concept descriptions for properties,
+    values and value lists conformant to IEC 61360.
+
+    :constraint AASd-010:
+        If DataSpecificationIEC61360/value is not empty then
+        DataSpecificationIEC61360/valueList shall be empty and vice versa.
```
[the remark this answers is not in the document]
Comment (mristin) [reply aas_core_meta/v3rc2.py:4703]: @g1zzm0 I'll add the invariants in the next step once we merge this PR.

File: aas_core_meta/v3rc2.py [hunk 2]
```diff
@@ -4530,36 +4689,48 @@ def __init__(
...
     """
     Preferred name
 
-    :constraint AASd-076:
-        For all ConceptDescriptions using data specification template
-        IEC61360
-        (http://admin-shell.io/DataSpecificationTemplates/DataSpecificationIEC61360/2/0)
-        at least a preferred name in English shall be defined.
+    :constraint AASc-002:
+        DataSpecification-IEC61360/preferredName shall be provided at least in English.
```
Comment (mristin) [inline aas_core_meta/v3rc2.py:4733]:
```suggestion
        :attr:`~preferred_name` shall be provided at least in English.
```

File: aas_core_meta/v3rc2.py [hunk 3]
```diff
@@ -4575,6 +4746,18 @@ class Data_specification_iec_61360(Data_specification_content):
...
+    unit and unitId need to be consistent if both attributes are set
+
+    .. note::
+
+        It is recommended to use a global reference.
+
+    .. note::
+
+        Although the unitId is a global reference there might exist a ConceptDescription
+        with data specification DataSpecificationPhysicalUnit with the same ID.
```
Comment (mristin) [inline aas_core_meta/v3rc2.py:4759]:
```suggestion
        Although the :attr:`~unit_id` is a global reference there might exist a 
        :class:`.Concept_description`
        with data specification :class:`.Data_specification_physical_unit` with
        the same ID.
```

**Review by mristin — CHANGES_REQUESTED** [review]
Thanks, @g1zzm0!

Most of my suggestions are references and refinement of the comments, so that they are a bit cleaner for the reader (short comments tend to be misread/omitted).

Let's first have others review this code and we merge in this pull request.

Afterwards, I'll write the invariants and double-check the code against the book once more.

**Comment by sebbader** [comment]
> @sebbader for the potential change of HasDataSpecification see the branch https://github.com/aas-core-works/aas-core-meta/tree/g1zzm0/EmbeddedDataSpecification

As written, fine with me to have it in another step 👍

**Comment by sebbader** [comment]
> @sebbader As discussed before, the schema generation is done after the validation of the metamodel or now the classes from chapter 6 and 7. The workflow will be Python MM Modeling -> Review MM -> Validation -> Code Generation -> New Schema Generation -> Check MM based on the serialization. This is necessary to avoid unnecessary correction loops and thus delays.

Then I regard my review as finished for now 👍 . Looks good to me but I am far away of being able to detect issues based on the python file 😄

**Comment by mristin** [comment]
@all please send your reviews till Tuesday 17h CET, so that we can continue with the changes and avoid conflicts. If there are more points, please open an issue.

File: aas_core_meta/v3rc2.py [hunk 4]
```diff
@@ -4447,33 +4455,179 @@ def __init__(
...
+    Documentation = "DOCUMENTATION"
+    Entity = "ENTITY"
+    Event = "EVENT"
+    Function = "FUNCTION"
+    Property = "PROPERTY"
+    Value = "VALUE"
+    Range = "RANGE"
+    Qualifier_type = "QUALIFIER_TYPE"
+    Referencing = "REFERENCING"
+    Relationship = "RELATIONSHIP"
```
[the remark this answers is not in the document]
Comment (g1zzm0) [reply aas_core_meta/v3rc2.py:4622]: For the sake of completeness, we will keep this in the initial 1:1 commit for now.

**Review by mristin — APPROVED** [review]

**Comment by mristin** [comment]
@g1zzm0 I reviewed your latest commits -- thanks, all LGTM!

**Pull request merged**